### PR TITLE
Full factorisation new name

### PR DESCRIPTION
**Pull request by Raphael-Tresor**
# Changes 
src/node.jl : FullFactorisation replaced by BetheFactorisation and FullFactorisation become depricated and an alias for BetheFactorisation.
test/node_tests.jl : duplicate basic test of FullFactorisation with BetheFactorisation.

### CODE REVIEW

**Comment by wouterwln** [comment]
I'd prefer `FullJoint` over `FullJoin`, since we're modeling a full joint probability distribution over all interfaces in a node.

**Comment by bvdmitri** [comment]
Thank you for working on this @Raphael-Tresor ! A couple of thoughts
- After you're done with the changes, you should call `make format` in the terminal in the package's folder to automatically adjust the formatting in the files. 
- I would probably also use the `@deprecate_binding` macro in Julia, see more info [here](https://invenia.github.io/blog/2022/06/17/deprecating-in-julia/) 
- Regarding the name I find the suggestion from @wouterwln good - `FullJoint`
- If we deprecate the binding its better to go all examples and documentation in RxInfer.jl repo and change the default to `FullJoin(t)`. But that can be a separate issue

File: src/node.jl [hunk 1]
```diff
@@ -125,27 +125,21 @@ as_node_symbol(fn::F) where {F <: Function} = Symbol(fn)
...
-See also: [`FullJoin`](@ref)
+See also: [`BetheFactorisation`](@ref)
 """
 struct MeanField end
 
 """
-    FullJoin
+    BetheFactorisation
 
 Generic factorisation constraint used to specify a full factorisation for recognition distribution `q`.
```
Comment (bvdmitri) [inline src/node.jl:135]:
```suggestion
Generic factorisation constraint used to specify the Bethe factorisation for recognition distribution `q`.
```

File: src/node.jl [hunk 2]
```diff
@@ -125,27 +125,21 @@ as_node_symbol(fn::F) where {F <: Function} = Symbol(fn)
...
 
-"""
-    FullFactorisation
-
-Alias for FullJoin. To depricate FullFactorisation.
-
-See also: [`FullJoin`](@ref)
-"""
-const FullFactorisation = FullJoin
+#Alias for `BetheFactorisation` to depricate `FullFactorisation`.
```
Comment (bvdmitri) [inline src/node.jl:141]:
```suggestion
# Alias for `BetheFactorisation` to deprecate `FullFactorisation`. 
```

**Review by bvdmitri — COMMENTED** [review]
Very nice @Raphael-Tresor , I spotted a couple of typos. The rest seems fine

**Comment by Raphael-Tresor** [comment]
> Very nice @Raphael-Tresor , I spotted a couple of typos. The rest seems fine

Thanks for the review @bvdmitri , I committed your corrections.

**Review by bvdmitri — APPROVED** [review]
Thanks @Raphael-Tresor !

**Pull request merged**